### PR TITLE
[SCB-394] framework version support SC which earlier than 1.0.0

### PR DESCRIPTION
**Pull request by weichao666**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by weichao666** [comment]
for the ServiceCenter's version of 1.0.0 and above,sdk report framework's version to SC,
for SC's version below 1.0.0, such as 0.5.0, sdk doesn't report framework's version

**Comment by weichao666** [comment]
image

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/api/registry/FrameworkVersions.java [hunk 1]
```diff
@@ -22,12 +22,29 @@
...
+  private static final Logger LOGGER = LoggerFactory.getLogger(FrameworkVersions.class);
   private static final ServiceLoader<Versions> frameworkVersions = ServiceLoader.load(Versions.class);
 
   public static String allVersions() {
+    ServiceRegistryClient client = RegistryUtils.getServiceRegistryClient();
+    ServiceCenterInfo serviceCenterInfo = client.getServiceCenterInfo();
+    if (serviceCenterInfo == null) {
+      LOGGER.error("query servicecenter version info failed.");
+    }
+    String scVersion = serviceCenterInfo.getVersion();
```
Comment (liubao68) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/api/registry/FrameworkVersions.java]: NPE

**Comment by wujimin** [comment]
1.save serviceCenterInfo in AbstractServiceRegistry, when connected to serviceCenter then fetch this information  
2.compare version by class "Version"

**Comment by wujimin** [comment]
oh. in pull mode, there is no "connected" event  
and then pull from LB can not  know talking to whick one SC instance.

so, maybe just make sure it's the first api send to SC

**Comment by wujimin** [comment]
not only this one feature need compatible logic  
there are more and more features need compatible logic in the furture  
so we should not ask all the features to query version from SC.

**Comment by liubao68** [comment]
We have a minimal supported SC constraints. So this code should be temporary and will be removed soon. But the mechanism can be left for products compatible requirements.

**Comment by weichao666** [comment]
image

**Comment by weichao666** [comment]
@wujimin modified

**Pull request merged**